### PR TITLE
hwCPUTemperature for NRF5 implementation

### PR DESCRIPTION
**Pull request by KooLru**
Get temperature via NRF_TEMP->TEMP
Tested on NRF52832 E73-2G4M04S1B module

### CODE REVIEW

File: hal/architecture/NRF5/MyHwNRF5.cpp [hunk 1]
```diff
@@ -523,7 +523,18 @@ uint16_t hwCPUFrequency(void)
 
 int8_t hwCPUTemperature(void)
 {
-	return -127; // not implemented yet
+	uint32_t Temperature = 0;
```
Comment (mfalkvidd) [inline hal/architecture/NRF5/MyHwNRF5.cpp]: Should this be signed? I guess we could get negative temperatures?

Comment (KooLru) [reply hal/architecture/NRF5/MyHwNRF5.cpp]: I fix Temperature type same as NRF_TEMP->TEMP type int32_t

**Review by tekka007 — CHANGES_REQUESTED** [review]
This implementation does not address anomalies present in certain NRF5x silicons. Have a look at https://github.com/xueliu/nRF52/blob/master/nRF52_SDK_0.9.1_3639cc9/examples/peripheral/temperature/main.c for reference. `nrf_temp_init()` and `nrf_temp_read()` are present in _hal\architecture\NRF5\drivers\nrf_temp.h_ and should be used instead.

File: hal/architecture/NRF5/MyHwNRF5.cpp [hunk 2]
```diff
@@ -523,7 +523,34 @@ uint16_t hwCPUFrequency(void)
...
+
+		NRF_TEMP->EVENTS_DATARDY = 0;
+
+		/**@note Workaround for PAN_028 rev2.0A anomaly 29 - TEMP: Stop task clears the TEMP register. */
+		Temperature += nrf_temp_read();
+
+		/**@note Workaround for PAN_028 rev2.0A anomaly 30 - TEMP: Temp module analog front end does not power down when DATARDY event occurs. */
+		NRF_TEMP->TASKS_STOP = 1; /** Stop the temperature measurement. */
+
+		wait(10);
```
Comment (tekka007) [inline hal/architecture/NRF5/MyHwNRF5.cpp]: you should use `delay()` here

**Review by tekka007 — CHANGES_REQUESTED** [review]

File: hal/architecture/NRF5/MyHwNRF5.cpp (hunk 2 above)
Comment (KooLru) [reply hal/architecture/NRF5/MyHwNRF5.cpp]: Fixed.

**Review by tekka007 — APPROVED** [review]

**Comment by tekka007** [comment]
Thanks for the contribution!

**Pull request merged**